modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/CmmnXmlConverter.java

```diff
@@ -479,101 +509,105 @@ protected void processPlanFragment(CmmnModel cmmnModel, PlanFragment planFragmen
 
     protected void processPlanItems(CmmnModel cmmnModel, PlanFragment planFragment) {
         for (PlanItem planItem : planFragment.getPlanItems()) {
-
             // Plan items are defined on the same level or in a higher parent stage, never in a child stage.
             Stage parentStage = planItem.getParentStage();
             PlanItemDefinition planItemDefinition = parentStage.findPlanItemDefinitionInStageOrUpwards(planItem.getDefinitionRef());
             if (planItemDefinition == null) {
                 throw new FlowableException("No matching plan item definition found for reference "
-                        + planItem.getDefinitionRef() + " of plan item " + planItem.getId());
+                    + planItem.getDefinitionRef() + " of plan item " + planItem.getId());
             }
+
             planItem.setPlanItemDefinition(planItemDefinition);
+            procesPlanItem(cmmnModel, planItem, planItemDefinition);
 
-            if (!planItem.getEntryCriteria().isEmpty()) {
-                resolveEntryCriteria(planItem);
-            }
+        }
 
-            if (!planItem.getExitCriteria().isEmpty()) {
-                boolean exitCriteriaAllowed = true;
-                if (planItemDefinition instanceof Task) {
-                    Task task = (Task) planItemDefinition;
-                    if (!task.isBlocking() && StringUtils.isEmpty(task.getBlockingExpression())) {
-                        exitCriteriaAllowed = false;
-                    }
-                }
+    }
 
-                if (exitCriteriaAllowed) {
-                    resolveExitCriteriaSentry(planItem);
-                } else {
-                    LOGGER.warn("Ignoring exit criteria on plan item {}", planItem.getId());
-                    planItem.getExitCriteria().clear();
+    protected void procesPlanItem(CmmnModel cmmnModel, PlanItem planItem, PlanItemDefinition planItemDefinition) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 532-547 into a separate method.